### PR TITLE
Consolidate generated 

### PR DESCRIPTION
**Pull request by sentilesdal**
This one got out of hand, sorry.  Here's what's included though.

-  Only defines structs once, they are imported correctly in other files.
- `hardhat` support added.
- `solc` multiple contracts per json support added.
- quiets the output of the formatters
- support for custom types from structs, no more warnings about structs when using `pypechain`

### CODE REVIEW

File: pypechain/hardhat/utilities.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+"""Utilities for working with foundry-rs."""
```
Comment (dpaiton) [inline pypechain/hardhat/utilities.py:1]: should this be "hardhat" or is "foundry-rs" correct? `hardhat/types.py` says "hardhat"

File: pypechain/render/contract.py [hunk 2]
```diff
@@ -19,14 +23,98 @@
...
+    """Get a dictionary of ContractInfos from an AbiInfos list.
+
+     This helps us organize structs and events by the contracts they are defined
+     in. Because structs can be defined in one file, and used in another, we
+     need to avoid duplication of the definitions.
+
+    Parameters
+    ----------
+    abi_infos : list[AbiInfo]
+        _description_
```
Comment (dpaiton) [inline pypechain/render/contract.py:53]: might as well just fill these in now, it's only going to get harder to do it later

File: pypechain/render/contract.py [hunk 3]
```diff
@@ -19,14 +23,98 @@
...
+    """_summary_
+
+    Parameters
+    ----------
+    events : EventInfo | list[EventInfo]
+        _description_
+    contract_name : str
+        _description_
+    """
+    if not isinstance(events, list):
```
Comment (dpaiton) [inline pypechain/render/contract.py:109]: does it make sense to check for Sequence instead of list? i.e. can one input a tuple?

**Comment by dpaiton** [comment]
I tried running this on agent0 and the `HyperdriveFactoryTypes` file uses the `Fees` struct, but does not import it.
image

**Comment by dpaiton** [comment]
Can you drop the python version to 3.10 in pyproject.toml, and change the "request" dependency to "requests"

**Review by dpaiton — CHANGES_REQUESTED** [review]
Looks good aside from some minor requests & the bug with importing Fees. Since it's such a big PR I'll mark it as "request changes" -- lmk when you've updated everything and I'll try running it against agent0 again.

File: pypechain/hardhat/utilities.py (hunk 1 above)
Comment (sentilesdal) [reply pypechain/hardhat/utilities.py:1]: yup yup

**Review by dpaiton — APPROVED** [review]

**Pull request merged**